src/idl_gen_fbs.cpp

```diff
@@ -15,12 +15,16 @@
  */
 
 // independent from idl_parser, since this code is not needed for most clients
-
 #include "flatbuffers/code_generators.h"
 #include "flatbuffers/flatbuffers.h"
 #include "flatbuffers/idl.h"
 #include "flatbuffers/util.h"
 
+#include <stdexcept>
```

Comment:
I agree with you 100%, and I throw exception here temporary, I prefer use [LogCompilerError](https://github.com/google/flatbuffers/blob/e1a2f688e0d12385fd977cb45d58453d88d7c0e2/include/flatbuffers/flatc.h#L32) but the problem is it defined in flatc.h and it needed to be included. On the other hand we needs to include GnerateFbs in flattest and it make trouble. So my plan is to separate Log function to  other file and clean them and then use LogCompilerError.

furthermore, [GenerateFBS](https://github.com/google/flatbuffers/blob/e1a2f688e0d12385fd977cb45d58453d88d7c0e2/src/idl_gen_fbs.cpp#L57) api needs to be changed, so that return boolean.

Both are under construction.